xline/src/storage/authstore/backend.rs

```diff
@@ -714,143 +715,131 @@ impl AuthStoreBackend {
+                Self::sync_role_get_request(&req);
             }
             RequestWrapper::AuthRoleGrantPermissionRequest(req) => {
                 debug!("Sync AuthRoleGrantPermissionRequest {:?}", req);
-                self.sync_role_grant_permission_request(req, next_revision)
+                self.sync_role_grant_permission_request(req);
             }
             RequestWrapper::AuthRoleRevokePermissionRequest(req) => {
                 debug!("Sync AuthRoleRevokePermissionRequest {:?}", req);
-                self.sync_role_revoke_permission_request(&req, next_revision)
+                self.sync_role_revoke_permission_request(&req);
             }
             RequestWrapper::AuthRoleListRequest(req) => {
                 debug!("Sync AuthRoleListRequest {:?}", req);
-                Self::sync_role_list_request(&req)
+                Self::sync_role_list_request(&req);
             }
             RequestWrapper::AuthRoleDeleteRequest(req) => {
                 debug!("Sync AuthRoleDeleteRequest {:?}", req);
-                self.sync_role_delete_request(&req, next_revision)
+                self.sync_role_delete_request(&req);
             }
             RequestWrapper::AuthenticateRequest(req) => {
                 debug!("Sync AuthenticateRequest {:?}", req);
-                Self::sync_authenticate_request(&req)
+                Self::sync_authenticate_request(&req);
             }
             _ => {
                 unreachable!("Other request should not be sent to this store");
             }
-        };
-        if modify {
-            *self.revision.lock() = next_revision;
         }
         self.header_gen.revision()
     }
 
     /// Sync `AuthEnableRequest` and return whether authstore is changed.
-    fn sync_auth_enable_request(&self, _req: &AuthEnableRequest, revision: i64) -> bool {
+    fn sync_auth_enable_request(&self, _req: &AuthEnableRequest) {
         if self.is_enabled() {
-            return false;
+            return;
         }
+        let revision = self.revision.next();
         self.put(AUTH_ENABLE_KEY.to_vec(), vec![1], revision, 0);
         self.enabled.store(true, AtomicOrdering::Release);
         self.create_permission_cache();
-        true
     }
 
     /// Sync `AuthDisableRequest` and return whether authstore is changed.
-    fn sync_auth_disable_request(&self, _req: &AuthDisableRequest, revision: i64) -> bool {
+    fn sync_auth_disable_request(&self, _req: &AuthDisableRequest) {
         if !self.is_enabled() {
-            return false;
+            return;
         }
+        let revision = self.revision.next();
         self.put(AUTH_ENABLE_KEY.to_vec(), vec![0], revision, 0);
         self.enabled.store(false, AtomicOrdering::Release);
-        true
     }
 
     /// Sync `AuthStatusRequest` and return whether authstore is changed.
-    fn sync_auth_status_request(_req: &AuthStatusRequest) -> bool {
-        false
-    }
+    fn sync_auth_status_request(_req: &AuthStatusRequest) {}
```

Comment:
What's the point of existence of this function? It has neither a return value nor a function body

Reply:
I guess @themanforfree wants to keep it so that all the sync_... functions have the same form.